code/modules/projectiles/guns/projectile/banannon.dm

```diff
@@ -0,0 +1,72 @@
+/obj/item/weapon/gun/banannon
+	name = "banannon"
+	desc = "The most fearsome weapon ever wielded by clown mercenaries."
+	icon = 'icons/obj/gun.dmi'
+	icon_state = "banannon"
+	item_state = "banannon"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guninhands_left.dmi', "right_hand" = 'icons/mob/in-hand/right/guninhands_right.dmi')
+	recoil = 0
+	slot_flags = SLOT_BELT
+	flags = FPRINT
+	w_class = W_CLASS_MEDIUM
+	fire_delay = 0
+	fire_sound = null
+	clumsy_check = 0
+	var/max_ammo = 10
+	var/current_ammo = 10
+
+/obj/item/weapon/gun/banannon/update_icon()
+	if(current_ammo >= max_ammo)
+		icon_state = initial(icon_state)
+	else if(current_ammo <= 0)
+		icon_state = "banannon_empty"
+	else
+		icon_state = "[initial(icon_state)]_[round((current_ammo/max_ammo) * 10)]"
+
+/obj/item/weapon/gun/banannon/examine(mob/user)
+	..()
+	to_chat(user, "<span class='info'>It has [current_ammo] round\s remaining.</span>")
+
+/obj/item/weapon/gun/banannon/afterattack(atom/A as mob|obj|turf|area, mob/living/user as mob|obj, flag, params, struggle = 0)
+	if(flag)
+		return //we're placing gun on a table or in backpack
+	if(harm_labeled >= min_harm_label)
+		to_chat(user, "<span class='warning'>A label sticks the trigger to the trigger guard!</span>")//Such a new feature, the player might not know what's wrong if it doesn't tell them.
+
+		return
+	if(!current_ammo)
+		return click_empty(user)
+	if(in_chamber)
+		qdel(in_chamber)
+		in_chamber = null
+	in_chamber = new /obj/item/projectile/bullet/sabonana(src)
+	if(Fire(A,user,params, "struggle" = struggle))
+		current_ammo--
+	update_icon()
+
+/obj/item/weapon/gun/banannon/Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0, struggle = 0, var/use_shooter_turf = FALSE)
+	if(istype(user, /mob/living))
+		var/mob/living/M = user
+		if(!clumsy_check(M) && prob(50))
+			to_chat(M, "<span class='danger'>[src] blows up in your face.</span>")
+			M.take_organ_damage(0,20)
+			M.drop_item(src, force_drop = 1)
+			qdel(src)
+			return 0
+	return ..()
+
+/obj/item/weapon/gun/banannon/process_chambered()
+	if(in_chamber)
+		return 1
+	return 0
+
+/obj/item/weapon/gun/banannon/attackby(obj/item/W, mob/user)
+	..()
+	if(istype(W, /obj/item/weapon/reagent_containers/food/snacks/grown/banana))
+		if(current_ammo >= max_ammo)
+			return
+		current_ammo++
+		playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 25, 1)
+		user.drop_item(W, force_drop = 1)
```

Comment:
How would you insert a banana into the gun if the banana is stuck to your hand? Why force this?